crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1355,6 +1380,105 @@ mod tests {
             ]
         );
     }
+
+    #[test]
+    fn test_jupyter_magic_help_end() {
+        let source = r"
+?foo?
```

Comment:
What's the expected output for `????` or any other double question sequence without any content.

Reply:
Oh I tested this locally but didn't add it in the test case. We transform it to the way IPython does. Note that the first two `?` is consumed by `MagicKind`:
```rust
Tok::MagicCommand {
    value: "??".to_string(),
    kind: MagicKind::Help2,
}
```